### PR TITLE
Bump to m7, Remove Sudo Pallet, Clean up Chain Spec

### PR DESCRIPTION
**Pull request by hayesgm**
This patch, first off, bumps us up to m8 and also removes the `sudo` pallet, which we have not used in a long time and no longer deserves to be part of the overall runtime.

Secondly, we remove the majority of data from the `testnet` chainspec and instead opt to add that data in `chains/build_spec.js` which properly pulls in data like tokens and Starports from the already existing files (i.e. compound-config and `./ethereum/networks/{network}.json`). We make sure these files _are_ checked in to the repo so can reference them quite easily (and since they should rarely change, they are not part of releases but instead just binary relics in the source control itself). This method should be clean and easy to add new nets (e.g. another test-net or mainnet) and should work fairly well with local test-nets if we want to add custom configuration for that.

Next, we remove `LockOld` and `LockCashOld` to match our new deployment.

Additionally, we clean up the `scripts/chains/alice.sh`-type scripts to be unified and start to get them more consistently working. We make sure they use temp directories which should obviate the need for frequent purges, as well.

Finally, we add `rpc.json` to releases since that is now considered a key build artifact per release.

### CODE REVIEW

File: chains/build_spec.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
+#!env node
```
Comment (hayesgm) [inline chains/build_spec.js:1]: This is a script that builds `chains/{chain}/chain-spec.js` and `chains/{chain}/chain-spec-raw.js` based on data from `chains/{chain}/chain-config.json` and other sources (e.g. `compound-config` and `ethereum/networks/ropsten.json`)

File: chains/testnet/chain-config.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
+{
```
Comment (hayesgm) [inline chains/testnet/chain-config.json:1]: This is the chain-specific info used to build the chain-spec. I try to limit the data that needs to be here, e.g. by pulling the actual token config from `compound-chain/networks/ropsten.json` where possible

File: ethereum-client/src/events.rs [hunk 3]
```diff
@@ -3,24 +3,8 @@ use codec::{Decode, Encode};
 use our_std::convert::TryInto;
 use our_std::RuntimeDebug;
 
-/**
- * Deprecation Notice: `LockOld` and `LockCashOld` may be purged after
- *                     version 1.3.1 is deployed to all networks.
- */
 #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
 pub enum EthereumEvent {
-    LockOld {
```
Comment (hayesgm) [inline ethereum-client/src/events.rs:12]: Removed these here for the new spec.

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -99,13 +126,15 @@ pub fn development_config() -> ChainSpec {
...
     testnet_genesis(
         // Initial PoA authorities
-        vec![
-            authority_keys_from_seed("Alice", "0xc77494d805d2b455686ba6a6bdf1c68ecf6e1cd7"),
-            authority_keys_from_seed("Bob", "0x435228f5ad6fc8ce7b4398456a72a2f14577d9cd"),
-        ],
-        // Sudo account
-        get_account_id_from_seed::<sr25519::Public>("Alice"),
-        true,
+        vec![],
```
Comment (hayesgm) [inline node/src/chain_spec.rs:129]: For testnet, we remove the tons of bad defaults we had in here.

File: runtime/Cargo.toml [hunk 5]
```diff
@@ -43,7 +43,6 @@ frame-system-benchmarking = { default-features = false, git = 'https://github.co
 pallet-aura = { default-features = false, git = 'https://github.com/compound-finance/substrate', branch = 'master' }
 pallet-grandpa = { default-features = false, git = 'https://github.com/compound-finance/substrate', branch = 'master' }
 pallet-randomness-collective-flip = { default-features = false, git = 'https://github.com/compound-finance/substrate', branch = 'master' }
-pallet-sudo = { default-features = false, git = 'https://github.com/compound-finance/substrate', branch = 'master' }
```
Comment (hayesgm) [inline runtime/Cargo.toml:46]: Sudo is removed.

File: runtime/src/lib.rs [hunk 6]
```diff
@@ -121,7 +121,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     spec_name: create_runtime_str!("gateway"),
     impl_name: create_runtime_str!("gateway"),
     authoring_version: 1,
-    spec_version: 6,
+    spec_version: 7,
```
Comment (hayesgm) [inline runtime/src/lib.rs:124]: Bump to m7

File: runtime/src/lib.rs [hunk 7]
```diff
@@ -382,7 +377,6 @@ construct_runtime!(
         Timestamp: pallet_timestamp::{Module, Call, Storage, Inherent},
         Aura: pallet_aura::{Module, Config<T>},
         Grandpa: pallet_grandpa::{Module, Call, Storage, Config, Event},
-        Sudo: pallet_sudo::{Module, Call, Config<T>, Storage, Event<T>},
```
Comment (hayesgm) [inline runtime/src/lib.rs:385]: Sudo is removed.

File: scripts/actors/__base.sh [hunk 8]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+set -e
```
Comment (hayesgm) [inline scripts/actors/__base.sh:3]: This script was copied and pasted three times and each had its own quirks. Here we just unify this script and make sure it works.

File: scripts/build_release.sh [hunk 9]
```diff
@@ -27,10 +27,11 @@ cargo +nightly build --release
 
 bin="./target/release/gateway"
 types="./types.json"
+rpc="./rpc.json"
 wasm="./target/release/wbuild/gateway-runtime/gateway_runtime.compact.wasm"
 
-if [ ! -f "$bin" -o ! -f "$wasm" -o ! -f "$types" -o ! -f "$contracts" ]; then
-	echo "Missing one of the following build files: $bin, $wasm, $types, $contracts"
+if [ ! -f "$bin" -o ! -f "$wasm" -o ! -f "$types" -o ! -f "$rpc" -o ! -f "$contracts" ]; then
```
Comment (hayesgm) [inline scripts/build_release.sh:33]: Add `rpc.json` to release artifacts.

**Review by jflatow — APPROVED** [review]

**Review by toninorair — APPROVED** [review]

**Pull request merged**